identity/auth-starter/src/main/java/io/camunda/identity/authentication/handler/CamundaAuthenticationFailureHandler.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.authentication.handler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.security.web.access.AccessDeniedHandler;
+import org.springframework.security.web.authentication.AuthenticationFailureHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+public final class CamundaAuthenticationFailureHandler
+    implements AuthenticationFailureHandler, AccessDeniedHandler, AuthenticationEntryPoint {
+
+  private final ObjectMapper objectMapper;
+
+  @Autowired
+  public CamundaAuthenticationFailureHandler(final ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public void onAuthenticationFailure(
+      final HttpServletRequest request,
+      final HttpServletResponse response,
+      final AuthenticationException error)
+      throws IOException {
+    handleFailure(request, response, HttpStatus.UNAUTHORIZED, error);
+  }
+
+  @Override
+  public void handle(
+      final HttpServletRequest request,
+      final HttpServletResponse response,
+      final AccessDeniedException error)
+      throws IOException {
+    // if a token was passed but could not be validated, onAuthenticationFailure is called
+    // however, if no token was passed, then access is denied here, and we want to distinguish
+    // between unauthorized and forbidden; we can do that by checking the session principal to see
+    // if it's authenticated or not
```

Comment:
I the past we have removed most comments from production code. Do you know how it is handled by zeebe/new monorepo code base?

Reply:
Good point, I left this in as I took it directly from the Zeebe so I assume its fine :)